server/src/weaverbird/backends/pypika_translator/translators/postgresql.py

```diff
@@ -50,5 +54,34 @@ def duration(
         )
         return StepContext(query, columns + [step.new_column_name])
 
+    def filter(
+        self,
+        *,
+        builder: 'QueryBuilder',
+        prev_step_name: str,
+        columns: list[str],
+        step: "FilterStep",
+    ) -> StepContext:
+
+        # To handle relative date formats
+        cond = step.condition
+        if isinstance(cond, (ConditionComboAnd, ConditionComboOr)):
+
+            operator = 'and_' if isinstance(cond, ConditionComboAnd) else 'or_'
+
+            for sub_cond in getattr(cond, operator) or []:
+                if isinstance(sub_cond, DateBoundCondition) and isinstance(
+                    getattr(sub_cond, 'value'), RelativeDate
+                ):
+                    setattr(sub_cond, 'value', evaluate_relative_date(getattr(sub_cond, 'value')))
+
```

Comment:
Why isn't this done in the base translator ?

Reply:
either one way or another, i though it will be better on postgres, i reversed that here :https://github.com/ToucanToco/weaverbird/pull/1413/commits/137bb7d0c51f1a851424f3bf03c93ac82dbe7bb8